common/utils/works.js

```diff
@@ -234,10 +246,27 @@ export function getItemIdentifiersWith(
   }, []);
 }
 
-export type ArchiveNode = {|
-  id: string,
-  title: string,
-  alternativeTitles: string[],
-  referenceNumber: string,
-  type: 'Work',
-|};
+export function getAncestorArray(work: Work): ArchiveNode[] {
```

Comment:
I'm not really sure I understand all the logic here. Are you looking for the ancestor with `partOf` as that is the nested one? You can just fetch the final item in the array if that is the case.

Reply:
That's the first part (which is only required while all the ancestors are present) but then it returns a flat array of the ancestors leading to the current work (like the original version of the API), as that's what the components expect.